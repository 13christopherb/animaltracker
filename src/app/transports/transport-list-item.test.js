import React from 'react';
import {shallow} from 'enzyme';
import TransportListItem from "./transport-list-item";
import moment from 'moment';
import toJson from 'enzyme-to-json';
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const transport = {
    departs: 'MBO',
    arrives: 'NRO',
    meetTime: moment('2017-09-15 09:30:00') //Arbitrary time
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<TransportListItem />', () => {

    describe('render()', () => {
        const store = mockStore()

        test('should render the component', () => {
            const wrapper = shallow(<TransportListItem store={store} transport={transport}/>);

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    });
});