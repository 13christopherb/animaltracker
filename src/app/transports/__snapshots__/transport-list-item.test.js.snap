// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransportListItem /> render() should render the component 1`] = `
<TransportListItem
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  transport={
    Object {
      "arrives": "NRO",
      "departs": "MBO",
      "meetTime": "2017-09-15T16:30:00.000Z",
    }
  }
/>
`;
